contracts/YieldStrategyLever.sol

```diff
@@ -0,0 +1,492 @@
+        bytes12 indexed vaultId,
+        bytes6 seriesId,
+        address indexed investor,
+        uint256 amountToInvest,
+        uint256 borrowAmount
+    );
+
+    event Repaid(
+        bytes6 ilkId,
+        bytes12 indexed vaultId,
+        bytes6 seriesId,
+        address indexed investor,
+        uint256 borrowAmountPlusFee,
+        uint256 ink,
+        uint256 art
+    );
+
+    event Closed(
+        bytes6 ilkId,
+        bytes12 indexed vaultId,
+        address indexed investor,
+        uint256 ink,
+        uint256 art
+    );
+
+    constructor(Giver giver_) {
+        giver = giver_;
+        IERC20(USDC).approve(
+            address(LADLE.joins(0x303200000000)),
+            type(uint256).max
+        );
+        IERC20(DAI).approve(
+            address(LADLE.joins(0x303100000000)),
+            type(uint256).max
+        );
+        IERC20(WETH).approve(
+            address(LADLE.joins(0x303000000000)),
+            type(uint256).max
+        );
+    }
+
+    // TODO: add auth when shipping
+    /// @notice Sets strategy for different ilks
+    /// @param ilkId Id of the ilk
+    /// @param strategy Strategy for the ilk
+    function setStrategy(bytes6 ilkId, IStrategy strategy) external {
+        strategies[ilkId] = strategy;
+        IERC20(CAULDRON.assets(ilkId)).approve(
+            address(LADLE.joins(ilkId)),
+            type(uint256).max
+        );
+    }
+
+    // TODO: add auth when shipping
+    /// @notice Approve maximally for an fyToken.
+    /// @param seriesId The id of the pool to approve to.
+    function approveFyToken(bytes6 seriesId) external {
+        IPool(LADLE.pools(seriesId)).fyToken().approve(
+            address(LADLE),
+            type(uint256).max
+        );
+    }
+
+    /// @notice Invest by creating a levered vault. The basic structure is
+    ///     always the same. We borrow FyToken for the series and convert it to
+    ///     the yield-bearing token that is used as collateral.
+    /// @param ilkId The ilkId to invest in. This is often a yield-bearing
+    ///     token, for example 0x303400000000 (WStEth).
+    /// @param seriesId The series to invest in. This series doesn't usually
+    ///     have the ilkId as base, but the asset the yield bearing token is
+    ///     based on. For example: 0x303030370000 (WEth) instead of WStEth.
+    /// @param amountToInvest The amount of the base to invest. This is denoted
+    ///     in terms of the base asset: USDC, DAI, etc.
+    /// @param borrowAmount The amount to borrow. This is denoted in terms of
+    ///     debt at maturity (and will thus be less before maturity).
+    /// @param fyTokenToBuy The amount of fyToken to be bought from the base
+    /// @param minCollateral Used for countering slippage. This is the minimum
+    ///     amount of collateral that should be locked. The debt is always
+    ///     equal to the borrowAmount plus flash loan fees.
+    function invest(
+        bytes6 ilkId,
```

Comment:
For this StrategyLever, though, we should call this parameter `strategyId`, right? It would be clearer.